hypertrace-core-graphql-span-schema/src/main/java/org/hypertrace/core/graphql/span/dao/SpanLogEventRequestBuilder.java

```diff
@@ -29,31 +33,67 @@
   private final FilterRequestBuilder filterRequestBuilder;
+  private final AttributeStore attributeStore;
 
   @Inject
   SpanLogEventRequestBuilder(
       Converter<Collection<AttributeRequest>, Set<Expression>> attributeConverter,
       Converter<Collection<AttributeAssociation<FilterArgument>>, Filter> filterConverter,
-      FilterRequestBuilder filterRequestBuilder) {
+      FilterRequestBuilder filterRequestBuilder,
+      AttributeStore attributeStore) {
     this.attributeConverter = attributeConverter;
     this.filterConverter = filterConverter;
     this.filterRequestBuilder = filterRequestBuilder;
+    this.attributeStore = attributeStore;
   }
 
   Single<LogEventsRequest> buildLogEventsRequest(
       SpanRequest gqlRequest, SpansResponse spansResponse) {
-    return zip(
-        this.attributeConverter.convert(gqlRequest.logEventAttributes()),
-        buildLogEventsQueryFilter(gqlRequest, spansResponse).flatMap(filterConverter::convert),
-        (selections, filter) ->
-            LogEventsRequest.newBuilder()
-                .setStartTimeMillis(
-                    gqlRequest.spanEventsRequest().timeRange().startTime().toEpochMilli())
-                .setEndTimeMillis(
-                    gqlRequest.spanEventsRequest().timeRange().endTime().toEpochMilli())
-                .addAllSelection(selections)
-                .setFilter(filter)
-                .build());
+    return getRequestAttributes(
+            gqlRequest.spanEventsRequest().context(), gqlRequest.logEventAttributes())
+        .flatMap(
+            v ->
+                zip(
+                    this.attributeConverter.convert(v),
+                    buildLogEventsQueryFilter(gqlRequest, spansResponse)
+                        .flatMap(filterConverter::convert),
+                    (selections, filter) ->
+                        LogEventsRequest.newBuilder()
+                            .setStartTimeMillis(
+                                gqlRequest
+                                    .spanEventsRequest()
+                                    .timeRange()
+                                    .startTime()
+                                    .toEpochMilli())
+                            .setEndTimeMillis(
+                                gqlRequest.spanEventsRequest().timeRange().endTime().toEpochMilli())
+                            .addAllSelection(selections)
+                            .setFilter(filter)
+                            .build()));
+  }
+
+  private Single<Collection<AttributeRequest>> getRequestAttributes(
+      GraphQlRequestContext requestContext, Collection<AttributeRequest> logEventAttributes) {
+    return this.attributeStore
+        .getForeignIdAttribute(
+            requestContext,
+            HypertraceCoreAttributeScopeString.LOG_EVENT,
+            HypertraceCoreAttributeScopeString.SPAN)
+        .flatMap(
+            spanId ->
+                logEventAttributes.stream()
```

Comment:
Since we're returning a collection anyway, can we just collect to a set rather than check if we already have it or not?

Reply:
We also could 1. probably simplify the nesting by just merging the two sources and 2. remove the nesting above by converting the attributes before returning. Something like:
```java
this.attributeStore
  .getForeignIdAttribute(
      requestContext,
      HypertraceCoreAttributeScopeString.LOG_EVENT,
      HypertraceCoreAttributeScopeString.SPAN)
  .map(attributeRequestBuilder::buildForAttribute) // can just inject this builder rather than create our own impl
  .toObservable()
  .mergeWith(Observable.fromIterable(logEventAttributes))
  .collect(Collectors.toUnmodifiableSet())
  .flatMap(this.attributeConverter::convert)
 ```